src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.InProcess.cs

```diff
@@ -177,9 +173,88 @@ internal abstract partial class AbstractNavigateToSearchService
             // well in the UI.
             var additionalMatchingProjects = await GetAdditionalProjectsWithMatchAsync(
                 document, declaredSymbolInfo, cancellationToken).ConfigureAwait(false);
-            return new SearchResult(
-                document, declaredSymbolInfo, kind, matchKind, isCaseSensitive, navigableItem,
-                matchedSpans.ToImmutable(), additionalMatchingProjects);
+            return new RoslynNavigateToItem(
+                isStale: false,
+                document.Id,
+                ComputeCombinedProjectName(document, additionalMatchingProjects),
+                declaredSymbolInfo.Name,
+                declaredSymbolInfo.NameSuffix,
+                declaredSymbolInfo.ContainerDisplayName,
+                declaredSymbolInfo.Kind,
+                declaredSymbolInfo.Accessibility,
+                declaredSymbolInfo.Span,
+                declaredSymbolInfo.IsPartial,
+                IsNonNestedNamedType(declaredSymbolInfo),
+                kind, matchKind,
+                isCaseSensitive,
+                matchedSpans.ToImmutable(),
+                ComputSecondarySort(declaredSymbolInfo));
```

Comment:
the type only holds onto the basic data that is needed to rehydrate the UI at the end.  importantly, it also doesn't need a Document to hold onto.  A doc-id is sufficiient.  This is important for a followup PR where we allow searchign just by doc-ids.